docs/faq/ImmutableData.md

```diff
@@ -1,245 +1,145 @@
 
-Fortunately, this can be easily overcome with the use of a browser extension such as the [Immutable.js Object Formatter](https://chrome.google.com/webstore/detail/immutablejs-object-format/hgldghadipiblonfkkicmgcbbijnpeog), which surfaces your data in Chrome Dev Tools, and hides Immutable.JS’s properties when inspecting your data.
+If, however, the returned value differs from the `state` object passed in (i.e. the `state` parameter passed in references a different object from that returned by the reducer), then Redux assumes a change has been made, and so will apply that change to the state tree.
 
-### Breaks object references, causing poor performance
+### How does React-Redux uses shallow equality checking?
+React-Redux shallowly compares the root state with the return values from the `mapStatToProps` function that is passed to its `connect` method in order to trigger the re-rendering of any affected components when the state is mutated.
 
-One of the key advantages of immutability is that it enables shallow equality checking, which dramatically improves performance. If two different variables reference the same immutable object, then a simple equality check of the two variables is enough to determine that they are equal, and that the object they both reference is unchanged. The equality check never has to check the values of any of the object’s properties, as it is, of course, immutable.
+If a shallow check of these values is true, React-Redux assumes the state has not changed, and so will not re-render the connected component.
 
-However, such shallow checking of object equality will not work if your data encapsulated within an Immutable.JS object is itself an object. This is because Immutable.JS’s `toJS()` method, which returns the data contained within an Immutable.js object as a JavaScript value, will create a new object every time it’s called, and so break the reference with the encapsulated data. Accordingly, calling `.toJS()` twice, for example, and assigning the result to two different variables will cause an equality check on those two variables to fail, even though the object values themselves haven’t changed.
+### Why will shallow equality checking not work with mutable objects?
+Shallow equality checking cannot be used to detect if a function mutates an object passed into it if that object is mutable.
 
-This, in turn, causes React’s `shouldComponentUpdate` method (and hence Redux’s `connect` function) to return true every time, causing an unchanged component to be re-rendered every time a change elsewhere in the app is detected, and so  severely degrading performance.
+This is self-evident if you think about it: if two variables reference the same object, they will _always_ be equal, regardless of whether the object changes or not, as they're both referencing the same object. Thus, the following will always return true:
```

Comment:
Think I'd like to see "self-evident" changed for another phrase.  Maybe:  "This is because of how objects and references work: if two..."